packages/core/src/App/Components/Layout/Header/menu-links.jsx

```diff
@@ -3,45 +3,79 @@ import React from 'react';
-                                    <Text
-                                        size='m'
-                                        line_height='xs'
-                                        title={item_text}
-                                        className='header__menu-link-text'
-                                    >
-                                        {item.icon}
-                                        {item_text}
-                                        {item.logo}
-                                    </Text>
-                                )}
-                                {item.image && (
-                                    <span className='header__menu-link-text'>
-                                        {item.image}
-                                        {item.logo}
-                                    </span>
-                                )}
-                            </React.Fragment>
-                        </BinaryLink>
-                    );
+                    return item.login_only && item.login_only !== is_logged_in
+                        ? null
+                        : (!is_pre_appstore && (
+                              <BinaryLink
+                                  id={item.id}
+                                  key={idx}
+                                  to={item.link_to || undefined}
+                                  onClick={item.onClick || undefined}
+                                  href={item.href || undefined}
+                                  className='header__menu-link'
+                                  active_class='header__menu-link--active'
+                              >
+                                  <React.Fragment>
+                                      {item_text && (
+                                          <Text
+                                              size='m'
+                                              line_height='xs'
+                                              title={item_text}
+                                              className='header__menu-link-text'
+                                          >
+                                              {item.icon}
+                                              {item_text}
+                                              {item.logo}
+                                          </Text>
+                                      )}
+                                      {item.image && (
+                                          <span className='header__menu-link-text'>
+                                              {item.image}
+                                              {item.logo}
+                                          </span>
+                                      )}
+                                  </React.Fragment>
+                              </BinaryLink>
+                          )) ||
+                              (is_pre_appstore && item_text !== 'Reports' && (
```

Comment:
This is too much nesting. Can you do an early return instead of nesting conditions like this? for better readability

Reply:
Will do